pyvista/plotting/helpers.py

```diff
@@ -189,8 +199,13 @@ def plot(var_item, off_screen=None, full_screen=False, screenshot=None,
                           jupyter_kwargs=jupyter_kwargs,
                           return_viewer=return_viewer)
 
-    # Result will be handled by plotter.show(): cpos or [cpos, img] or
-    # the jupyterlab scene when return_viewer is True
+    # show will not return camera position, get this from the plotter
+    if return_cpos:
+        cpos_out = plotter.camera_position
+        if result is not None:
+            return cpos_out, result
+        return cpos_out
+
```

Comment:
Allowing `show()` to `return_cpos` would mean that we can just `return result`, deferring the choice to `show()`.